src/check/runner/Sampler.ts

```diff
@@ -16,10 +14,10 @@ import { pathWalk } from './utils/PathWalker';
 function toProperty<Ts>(
   generator: IRawProperty<Ts> | Arbitrary<Ts>,
   qParams: QualifiedParameters<Ts>
-): INextRawProperty<Ts> {
+): IRawProperty<Ts> {
   const prop = !Object.prototype.hasOwnProperty.call(generator, 'isAsync')
     ? new Property(generator as Arbitrary<Ts>, () => true)
-    : convertToNextProperty(generator as IRawProperty<Ts>);
+    : (generator as IRawProperty<Ts>);
```

Comment:
This line is probably the reason of the issue #2736 reported by @nth-commit. I'll either patch it in v3 and v2 or just v3 🤔 But the `as` is definitely the source of the issue.